Public/Src/Engine/Processes/FileAccessManifest.cs

```diff
@@ -524,6 +817,12 @@ public ArraySegment<byte> GetPayloadBytes(FileAccessSetup setup, MemoryStream st
         [SuppressMessage("Microsoft.Usage", "CA2202:Do not dispose objects multiple times")]
         public byte[] GetManifestTreeBytes()
         {
+            if (IsManifestTreeBlockSealed)
+            {
+                Contract.Assert(m_sealedManifestTreeBlock != null);

```

Comment:
>t.Assert(m_sea [](start = 23, length = 14)

These asserts are crazy since they are teh same check as the boolean, so why have the boolean at all?

Reply:
Clearly you don't trust the property :)

---
In reply to: [276815332](https://github.com/Microsoft/BuildXL/pull/159#discussion_r276815332) [](ancestors = 276815332)

Reply:
I introduced property as a  name for that condition. Somehow, it's a reflex when I added that contract :-)

---
In reply to: [276815368](https://github.com/Microsoft/BuildXL/pull/159#discussion_r276815368) [](ancestors = 276815368,276815332)